api/v4/openapi.yaml

```diff
@@ -327,4 +333,12 @@ components:
         reportConfidence:
           type: string
         cvss:
-          type: string
\ No newline at end of file
+          type: string
+    RootComponent:
```

Comment:
I would add this as a new provider `redhat` otherwise this will break the general data hierarchy. These recommendations come from some provider and maybe it eventually might include not only recommendations but also vulnerabilities and remediations for such packages.

Reply:
I'll remove the `recommendation` part and use a new provider instead.

But I need to store the purl of the original sbom somewhere. The purl will be used to match the image names in the docker file and also to show the image name in the html report. 
@ruromero where do you think is a good place to store the purl? Also in the `redhat` provider?

Reply:
The `DependencyReport` has a `ref` which should be the original purl and the `recommendation` which is the purl of what you are recommending. Is that what you are asking?

Reply:
Yes, that's what I am asking.

In case there are multiple `DependencyReport` from provider `redhat`, how could we determine which `DependencyReport` is for the root component of the request `sbom`? The one without any `issues`?

Reply:
I understand. But the main purpose of adding the `purl` is to create an `id` for the `AnalysisReport`, with which we can match the `AnalysisReport` with an sbom in the array of the request body, and also distinguish different `AnalysisReport` in the html report (e.g. use the `id` as tag title in the html report). 

If we store the `purl` as a dependency in provider/source and don't create an `id` for the `AnalysisReport`, how should we match the sbom and generate tag titles for the html report? Should we use the position in the array to match the `AnalysisReport` to an sbom from the input array? Should we create a new variable to the html report for tag titles instead of getting the titles from the `AnalysisReport`?

Also, if we store the `purl` as a dependency in provider/source, then each provider would have at least one recommendation for the image. In the meantime, we just want to emphasize the single recommendation for UBI.
In this case, should we just get the first recommendation from any provider and assume it will always be the UBI that we want to recommend?

Reply:
what if you make the response in the batch-analysis a dictionary instead of a list?

Reply:
You can, as you say rely on the array order or turn the list into a dictionary

Reply:
Another issue is that the root component is not really a dependency, it is the application itself.
If we add the root component as a `DependencyReport` in the `Source`,  should all other dependencies become transitive?
And how should we handle the root component in the `SourceSummary` and in the html report? Ignore it?
In addition, for snyk, `oci` packages are not supported. So the image will be treated as an unscanned package, and there will not be a `DependencyReport` with recommendations for the image.

Should we create a new property for the root component in the `Source`, instead of adding it to the `DependencyReport` list?

Reply:
An SBOM is a bill of materials, it doesn't necessarily need to be a tree. In our case the SBOM will contain the image + rpms + python packages + etc. there is no dependency tree. They will all be components at the same level, then of course we will generate a tree for the python transitive deps etc.
Does that make sense? Maybe I'm missing something.